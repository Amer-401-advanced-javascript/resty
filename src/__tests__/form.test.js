import React from 'react';

import {shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';
import Form from '../components/form/form.js';



describe('<Form />', () => {
    it('Form Exists', () => {
        let app = shallow(<Form />);
        expect(app.find('form').exists()).toBeTruthy();
    });
    
    it('generate url and method', () => {
           let app = mount(<Form />);
           let input = app.find('#url');
           let get = app.find('#get');
           let form = app.find('form');

           input.simulate('change', {target:{value:'local'}});
           get.simulate('click');

           expect(app.state('url')).toBe('local');
           expect(app.state('method')).toBe('GET');

           form.simulate('submit');
           expect(app.state('url')).toBe('');
           expect(app.state('method')).toBe('');
    });



})