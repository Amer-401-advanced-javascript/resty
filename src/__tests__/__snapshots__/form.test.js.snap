// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> render correctly 1`] = `
Array [
  <form
    onSubmit={[Function]}
  >
    <label
      className="url"
    >
       URL
    </label>
    <input
      id="url"
      onChange={[Function]}
      type="text"
    />
    <button
      type="submit"
    >
      GO!
    </button>
    <label
      for="get"
    >
      GET
    </label>
    <input
      id="get"
      name="rest"
      onClick={[Function]}
      type="radio"
      value="GET"
    />
    <label
      for="post"
    >
      POST
    </label>
    <input
      id="post"
      name="rest"
      onClick={[Function]}
      type="radio"
      value="POST"
    />
    <label
      for="put"
    >
      PUT
    </label>
    <input
      id="put"
      name="rest"
      onClick={[Function]}
      type="radio"
      value="PUT"
    />
    <label
      for="delete"
    >
      DELETE
    </label>
    <input
      id="delete"
      name="rest"
      onClick={[Function]}
      type="radio"
      value="DELETE"
    />
  </form>,
  <section
    id="result"
  >
    <span
      className="method"
    />
    <span
      className="url"
    />
  </section>,
]
`;
